src/js/components/Form/Form.js

```diff
@@ -31,7 +31,7 @@ const stringToArray = (string) => {
   return undefined;
 };
 
-const getValueForName = (name, value) => {
+const getFieldValue = (name, value) => {
```

Comment:
I like the naming changes and the test changes. I think there are enough of them that it might be easier to digest the changes if they were separated out into a separate pull request.